src/test/ui/destructuring-assignment/tuple_destructure.rs

```diff
@@ -16,4 +16,7 @@ fn main() {
     assert_eq!((a, b), (2, 2));
     (b, ..) = (5, 6, 7);
     assert_eq!(b, 5);
+
+    // Check the empty tuple.
+    () = ();
```

Comment:
This is quite amusing.

Reply:
since the assignment expression returns `()` this means the following is now valid Rust:

```rust
()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=()=();
```

Reply:
A beautiful addition to the [`weird-exprs`](https://github.com/rust-lang/rust/blob/master/src/test/ui/weird-exprs.rs) test.

Reply:
Add it to the `weird-exprs.rs` test :)

Reply:
Perhaps:
```rust
(..)=(,)=()=(,)=(..);
```